tests/access_test.go

```diff
@@ -0,0 +1,158 @@
+package tests
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	authorization "k8s.io/api/authorization/v1"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	commonTemplates "kubevirt.io/ssp-operator/internal/operands/common-templates"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = FDescribe("[rfe_id:500]Golden Images RBAC", func() {
+	BeforeEach(func() {
+		namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testServiceAccountNS}}
+		Expect(apiClient.Create(ctx, namespaceObj)).ToNot(HaveOccurred(), "Failed to create namespace")
+
+		saviewObj := &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: testViewServiceAccountName, Namespace: testServiceAccountNS}}
+		Expect(apiClient.Create(ctx, saviewObj)).ToNot(HaveOccurred(), "Failed to create ServiceAccount")
+
+		waitUntilDeployed()
+	})
+	AfterEach(func() {
+		saviewObj := &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: testViewServiceAccountName, Namespace: testServiceAccountNS}}
+		err := apiClient.Delete(ctx, saviewObj)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testServiceAccountNS}}
+		err = apiClient.Delete(ctx, namespaceObj)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		waitForDeletion(client.ObjectKey{
+			Name: testServiceAccountNS,
+		}, &v1.Namespace{})
+	})
+
+	table.DescribeTable("should verify resource permissions",
+		func(resourceAttr *authorization.ResourceAttributes, expectedFail bool) {
+			sar := authorization.SubjectAccessReview{
+				Spec: authorization.SubjectAccessReviewSpec{
+					User: fmt.Sprintf("system:serviceaccount:%s:%s", testServiceAccountNS, testViewServiceAccountName),
+					Groups: []string{
+						"system:serviceaccounts",
+						"system:serviceaccounts:" + testServiceAccountNS,
+						"system:authenticated",
+					},
+					ResourceAttributes: resourceAttr,
+				},
+			}
+			sar.Spec.ResourceAttributes.Namespace = commonTemplates.GoldenImagesNSname
+			resultSar, err := SendSubjectAccessReview(&sar)
+			result := resultSar.Status.Allowed
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Please swap these 2 lines:
```
result := resultSar.Status.Allowed
Expect(err).ToNot(HaveOccurred())
```

It is a golang convention, that if `err` is not `nil`, then the returned value is `nil`. So ` resultSar.Status.Allowed` would cause NPE.